### PR TITLE
WIP - Allows Stringy to work without ext-mbstring

### PR DESCRIPTION
**Pull request by lucasmichot**
Allows Stringy to work without ext-mbstring

### CODE REVIEW

**Comment by danielstjules** [comment]
It'd be nice to allow users the freedom to use either the extension or polyfill.

**Comment by danielstjules** [comment]
As-is, I don't think I'd merge, as I suspect the polyfill is quite a bit slower than the extension. I'd have to benchmark first. But it would be great to allow users to choose from one or the other.

**Comment by GrahamCampbell** [comment]
> It'd be nice to allow users the freedom to use either the extension or polyfill.

User DO have the freedom. The poylfill is a fallback for when the extension is not avialble. It does not replace it.

**Comment by danielstjules** [comment]
I realize that it's a fallback. My comment was in regards to it still being a forced dependency, rather than a suggested one. From a different perspective, I don't deliver shims/polyfills for IE9 to latest chrome - it's unnecessary.

**Comment by GrahamCampbell** [comment]
> As is forcing anyone with ext-mbstring to download this polyfill.

Yes, ture, but that's not that much of a hardship imo.

**Comment by danielstjules** [comment]
I think I'd be more open to it if I could figure out how to get travis building all versions of PHP with ext-mbstring enabled/disabled - doubling up on the number of jobs.

**Comment by danielstjules** [comment]
Sweet! Let's see how this runs :)

**Comment by lucasmichot** [comment]
Seems we both are hanged on https://travis-ci.org/danielstjules/Stringy/builds/108171907 at the same time @danielstjules  ;-)

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
+
 php:
-  - 5.6
-  - 5.5
-  - 5.4
   - 5.3
+  - 5.4
+  - 5.5
+  - 5.6
   - 7.0
```
Comment (danielstjules) [inline .travis.yml]: Could all of the versions above be removed so that we're just left with `- hhvm`? Won't all others be ran as a result of the matrix?

Comment (lucasmichot) [inline .travis.yml]: See https://github.com/danielstjules/Stringy/commit/c3f25577ed8ef1d1d78f26ef182359182187280a

**Comment by danielstjules** [comment]
Looks good, thanks again! :)

**Pull request merged**